### PR TITLE
feat: hub identity and config endpoints (CO1 2.3, 2.4)

### PR DESCRIPTION
**Pull request by mikejgray**
## Summary

- **Hub Identity (CO1 2.3):** `GET /hub/identity` returns a stable, memorable hub ID (e.g. `bold-crystal-synapse`), user-configurable display name, and software version. `POST /hub/identity` (auth required) updates the display name. Hub ID is generated once at first boot and persisted to config.
- **Hub Config (CO1 2.4):** `GET /hub/config` returns the active TTS, STT, and LLM engine names. TTS/STT are read from `neon.yaml` with known defaults (`neon-tts-plugin-coqui`, `neon-stt-plugin-nemo`) as fallback. LLM is hardcoded to "Neon Classic". Gracefully handles missing, malformed, or unreadable `neon.yaml` for non-Hub deployments.
- Both discovery endpoints are public (no auth) so Nodes can identify and inspect a Hub during network discovery.

## Files changed

- `neon_hana/hub_id.py` — Docker-style 3-word kebab-case ID generator (new)
- `neon_hana/schema/hub_requests.py` — Pydantic v2 request/response models (new)
- `neon_hana/app/routers/hub.py` — Hub router with identity + config endpoints (new)
- `neon_hana/app/__init__.py` — Register hub router
- `tests/test_hub_id.py` — Generator unit tests (new)
- `tests/test_app.py` — 16 new endpoint and edge case tests

## Test plan

- [ ] `GET /hub/identity` returns 200 with non-empty `hub_id`, `display_name`, `version`
- [ ] `hub_id` is stable across requests and container restarts
- [ ] `POST /hub/identity` with valid auth updates display name (whitespace stripped)
- [ ] `POST /hub/identity` without auth returns 401/403
- [ ] `GET /hub/config` returns TTS/STT module names from neon.yaml
- [ ] `GET /hub/config` returns defaults when neon.yaml has no TTS/STT keys
- [ ] `GET /hub/config` returns null TTS/STT when neon.yaml is absent
- [ ] All 45 tests pass in Docker (`ghcr.io/neongeckocom/neon-hana:dev`)

### CODE REVIEW

**Comment by mikejgray** [comment]
**Note on CI failure:** The test failure is pre-existing and unrelated to this PR. `test_mq_websocket_api.py` fails during collection because `ovos_utils.signal` was removed upstream in a newer `ovos-utils` release:

```
neon_utils/file_utils.py:37: in <module>
    from ovos_utils.signal import ensure_directory_exists
E   ModuleNotFoundError: No module named 'ovos_utils.signal'
```

The import chain is `test_mq_websocket_api.py` → `MQWebsocketAPI` → `neon_iris.client` → `neon_utils.file_utils` → `ovos_utils.signal`. Last successful `dev` CI run was February 26 — the dependency break happened sometime after that.

All 45 tests in this PR pass locally in Docker (`ghcr.io/neongeckocom/neon-hana:dev`).

File: neon_hana/app/routers/hub.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    TTS/STT are read from neon.yaml on each request so that
+    config changes are reflected without restarting HANA.
+    If the file is not present (non-Hub deployment), those
+    fields are null. If present but the keys are not set,
+    known defaults are returned.
+
+    This endpoint is public (no authentication required) so that
+    Nodes can display Hub configuration during discovery.
+    """
+    neon_config = _read_neon_yaml()
```
Comment (NeonDaniel) [inline neon_hana/app/routers/hub.py:163]: As noted, this may fail if HANA doesn't share a local FS with the core services. Consider a TODO comment or issue to add Messagebus handlers as an alternative to get stt/tts config

File: neon_hana/schema/hub_requests.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+class STTConfig(BaseModel):
+    """STT plugin configuration."""
+
+    module: str = Field(description="Active STT plugin module name")
+
+
+class LLMConfig(BaseModel):
+    """LLM configuration."""
+
+    name: str = Field(description="Active LLM engine name")
```
Comment (NeonDaniel) [inline neon_hana/schema/hub_requests.py]: `LLMPersona` from https://github.com/NeonGeckoCom/neon-data-models/blob/dev/neon_data_models/models/api/llm.py#L54 should be a drop-in replacement for this class and will match endpoint returns

**Review by NeonDaniel — COMMENTED** [review]
Couple minor suggestions. I still need to deploy and test this before approving

File: neon_hana/app/routers/hub.py (hunk 1 above)
Comment (mikejgray) [reply neon_hana/app/routers/hub.py:163]: Will do. I started by looking for a message bus option, but oddly enough it isn't consistent enough to rely on

**Comment by NeonDaniel** [comment]
RE test failures and container errors, adding `neon-utils>=1.14.1a1` to `requirements.txt` should resolve the import error.

File: neon_hana/app/routers/hub.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+hub_route = APIRouter(prefix="/hub", tags=["hub"])
+
+# Generate hub_id eagerly at import time to avoid a race condition
+# where two concurrent first-boot requests each generate a different ID.
+_hub_id = config.get("hub_id")
+if not _hub_id:
+    _hub_id = generate_hub_id()
+    LOG.info("Generated new hub_id: %s", _hub_id)
+    update_mycroft_config({"hana": {"hub_id": _hub_id}})
```
Comment (NeonDaniel) [inline neon_hana/app/routers/hub.py]: This fails for k8s deployments with:
```
Traceback (most recent call last):                                                                                                                                                                                                                           │
│   File "/app/neon_hana/app/__main__.py", line 35, in <module>                                                                                                                                                                                                │
│     from neon_hana.app import create_app                                                                                                                                                                                                                     │
│   File "/usr/local/lib/python3.9/site-packages/neon_hana/app/__init__.py", line 38, in <module>                                                                                                                                                              │
│     from neon_hana.app.routers.hub import hub_route                                                                                                                                                                                                          │
│   File "/usr/local/lib/python3.9/site-packages/neon_hana/app/routers/hub.py", line 58, in <module>                                                                                                                                                           │
│     update_mycroft_config({"hana": {"hub_id": _hub_id}})                                                                                                                                                                                                     │
│   File "/usr/local/lib/python3.9/site-packages/ovos_config/config.py", line 403, in update_mycroft_config                                                                                                                                                    │
│     conf.store()                                                                                                                                                                                                                                             │
│   File "/usr/local/lib/python3.9/site-packages/ovos_config/models.py", line 138, in store                                                                                                                                                                    │
│     with open(path, 'w+', encoding="utf-8") as f:                                                                                                                                                                                                            │
│ OSError: [Errno 30] Read-only file system: '/config/neon/diana.yaml'                                                                                                                                                                                         │
│ stream closed EOF for alpha/endpoint-hana-54b74c7cf8-9g9rh (endpoint-hana)
```

In general, the hub_id should be set in config anyway, but consider gracefully handling config update failures.

```suggestion
    try:
        update_mycroft_config({"hana": {"hub_id": _hub_id}})
    except Exception as e:
        LOG.error(f"Failed to write config update: {e}")
```

File: neon_hana/app/routers/hub.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+async def update_hub_identity(
+    request: UpdateHubIdentityRequest,
+) -> HubIdentityResponse:
+    """
+    Update the display name of this Hub.
+
+    Requires authentication. The new display name is persisted to
+    the configuration file and survives container restarts.
+    """
+    update_mycroft_config({"hana": {"hub_display_name": request.display_name}})
```
Comment (NeonDaniel) [inline neon_hana/app/routers/hub.py]: This should be limited with RBAC. The `get_user` endpoint shows one way to handle the input token as a kwarg

```suggestion
    permissions = PermissionsConfig.from_roles(
        client_manager.get_token_data(token).roles)
    if permissions.hub < AccessRoles.ADMIN:
        raise HTTPException(status_code=403, detail=f"Insufficient Hub management permission ({permissions.hub})")
    update_mycroft_config({"hana": {"hub_display_name": request.display_name}})
```

**Review by NeonDaniel — CHANGES_REQUESTED** [review]
Deployed and tested at `hana.neonaialpha.com`. Noted a permissions check to implement and some exceptions I ran into that might be handled gracefully

**Review by NeonDaniel — APPROVED** [review]

File: neon_hana/auth/client_manager.py
Comment (NeonDaniel) [inline neon_hana/auth/client_manager.py:251]: Missing docstring for `node_auth`

**Review by NeonDaniel — CHANGES_REQUESTED** [review]

**Review by NeonDaniel — APPROVED** [review]
